libcontainer/factory_linux.go

```diff
@@ -159,13 +159,27 @@ func (l *LinuxFactory) Create(id string, config *configs.Config) (Container, err
 	}
 	containerRoot := filepath.Join(l.Root, id)
 	if _, err := os.Stat(containerRoot); err == nil {
-		return nil, newGenericError(fmt.Errorf("Container with id exists: %v", id), IdInUse)
+		pid, err := ioutil.ReadFile(filepath.Join(containerRoot, "task"))
+		if err == nil {
+			procPath := fmt.Sprintf("/proc/%s/cmdline", pid)
+			if cmdline, err := ioutil.ReadFile(procPath); err == nil {
+				_, exeName := filepath.Split(string(cmdline))
+				exeName = exeName[0:4]
+				if exeName[0:4] == "runc" {
```

Comment:
I'm not sure I'm following you. I'm saying at line 168 exeName is equal to exeName[0:4] and `len(exeName) == 4` because you set  `exeName = exeName[0:4]` the line before. In the same way, for any string s `s == s[0:len(s)]`. This code would function the exact same way without line 167.